lib/core/utils/contains.js

```diff
@@ -6,41 +6,32 @@
  * @param  {VirtualNode} otherVNode The vNode to test is contained by `vNode`
  * @return {Boolean}           Whether `vNode` contains `otherVNode`
  */
-function contains(vNode, otherVNode) {
+export default function contains(vNode, otherVNode) {
   /*eslint no-bitwise: 0*/
-
-  function containsShadowChild(vNode, otherVNode) {
-    if (vNode.shadowId === otherVNode.shadowId) {
-      return true;
-    }
-    return !!vNode.children.find(child => {
-      return containsShadowChild(child, otherVNode);
-    });
```

Comment:
This was the biggest problem. Searching all descendants of vNode instead of just looking at the ancestors of ovetherVNode.